packages/vite/src/node/plugins/html.ts

```diff
@@ -245,6 +245,8 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
               everyScriptIsAsync &&= isAsync
               someScriptsAreAsync ||= isAsync
               someScriptsAreDefer ||= !isAsync
+            } else if (url && path.extname(url).match(/\.([mc])?ts/) != null) {
```

Comment:
should probably check for any non-`.js` extension. for example, `.svelte` needs preprocessing

```ts
url && !url.match(/\.[mc]?js$/)
```